### PR TITLE
Fix a crash on ``namedtuples`` that use ``typename``

### PR DESCRIPTION
**Pull request by DanielNoord**
## Steps

- [x] For new features or bug fixes, add a ChangeLog entry describing what your PR does.
- [x] Write a good description on what the PR does.

## Description

Closes https://github.com/PyCQA/pylint/issues/7429

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

@Pierre-Sassoulas I couldn't get the inference failure path to cover, I think it is covered in another path.

Anyway I think it is good to have there, even if uncovered.

### CODE REVIEW

File: astroid/brain/brain_namedtuple_enum.py [hunk 1]
```diff
@@ -538,10 +538,22 @@ def _get_namedtuple_fields(node: nodes.Call) -> str:
     extract a node from them later on.
     """
     names = []
+    container = None
     try:
         container = next(node.args[1].infer())
     except (InferenceError, StopIteration) as exc:
         raise UseInferenceDefault from exc
+    # We pass on IndexError as well try to infer 'field_names' from the keywords
```
Comment (DanielNoord) [inline astroid/brain/brain_namedtuple_enum.py:546]:
```suggestion
    # We pass on IndexError as we'll try to infer 'field_names' from the keywords
```

File: ChangeLog [hunk 2]
```diff
@@ -17,7 +17,9 @@ What's New in astroid 2.12.9?
 =============================
 Release date: TBA
 
+* Fixed a crash on ``namedtuples`` that use ``typename`` to specify their name.
```
Comment (Pierre-Sassoulas) [inline ChangeLog:21]:
```suggestion


```
Keeping the number of line will prevent conflict later on because the diff will be contiguous.

**Review by Pierre-Sassoulas — APPROVED** [review]
👍

File: ChangeLog [hunk 3]
```diff
@@ -17,8 +17,11 @@ What's New in astroid 2.12.9?
...
```
Comment (Pierre-Sassoulas) [inline ChangeLog:21]:
```suggestion
```

File: ChangeLog [hunk 4]
```diff
@@ -17,8 +17,11 @@ What's New in astroid 2.12.9?
...
 
 
+  Closes PyCQA/pylint#7429
+
```
Comment (Pierre-Sassoulas) [inline ChangeLog:24]:
```suggestion


```

**Pull request merged**